### PR TITLE
ticket-1577 IMA Agent moved into it's own repository

### LINKED ISSUE #1577: Move IMA agent to a separate repo
Current repo structure holds in one repo:
- IMA mainnet contracts
- IMA predeployed contracts
- IMA agent

To make IMA to have more modular structure we need to retrieve IMA agent into separate repo and set up the separate versioning system for it

### LINKED ISSUE #1579: Minimize NPM dependencies and memory usage of IMA Agent
Source code of IMA Agent uses many same and big NPM modules installed in different folders. These dependencies can be united into one folder what can minimize disk space and RAM usage up to 3x times.

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
This pull request is massive refactoring in IMA.  IMA Agent moved into it's own repository.
No new tests needed. Existing tests were split between two repositories: `IMA` and new `ima-agent`.
Also includes improvements to minimize disk space occupied by NPM modules as dependencies (https://github.com/skalenetwork/IMA/issues/1579).

### CODE REVIEW

**Comment by sergiy-skalelabs** [comment]
Also includes improvements to minimize disk space occupied by NPM modules as dependencies (https://github.com/skalenetwork/IMA/issues/1579).

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* @sergiy-skalelabs @kladkogex
+*.md @skalenetwork/docowners
+/proxy/ @payvint
```
Comment (DmytroNazarenko) [inline CODEOWNERS]: Update it, please

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[submodule "IMA"]
+	path = IMA
+	#url = git@github.com:skalenetwork/IMA.git
```
Comment (DmytroNazarenko) [inline .gitmodules]: Remove comments, please

File: README.md [hunk 3]
```diff
@@ -1,2 +1,64 @@
...
-Containerized SKALE Interchain Messaging Agent
+<!-- SPDX-License-Identifier: (AGPL-3.0-only OR CC-BY-4.0) -->
+
+# SKALE Interchain Messaging Agent (IMA)
+
+[![Discord](https://img.shields.io/discord/534485763354787851.svg)](https://discord.gg/vvUtWJB)
+
+## A critical note about production readiness
+
+The IMA is still in active development and therefore should be regarded as _alpha software_. The development is still subject to further security hardening, testing, and breaking changes.
```
Comment (DmytroNazarenko) [inline README.md]: We can remove it

File: README.md [hunk 4]
```diff
@@ -1,2 +1,64 @@
...
+## Software Architecture
+
+IMA consists of the following three parts:
+
+1)  `Mainnet` smart contracts.
+2)  `SKALE Chain` smart contracts.
+3)  A containerized Agent application.
+
+Smart contracts are interfaces for any software working with `Mainnet` and `SKALE Chain` like other smart contracts deployed there or software connecting these Ethereum networks.
+The Agent is a Node JS application connecting the smart contracts on Mainnet with SKALE Chains.
```
Comment (DmytroNazarenko) [inline README.md]: Describe agent architecture only here

File: README.md [hunk 5]
```diff
@@ -1,2 +1,64 @@
...
+
+[SKALE Network Browser](npms/skale-observer/README.md). Responsible for providing description of all SKALE chains.
+
+#### IMA Log
+
+Console and log file output with rotation.
+
+#### IMA CC
+
+ANSI colorizer for console and log output.
```
Comment (DmytroNazarenko) [inline README.md]: Is this still relevant?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "skale-ima-agent",
+    "private": true,
```
Comment (DmytroNazarenko) [inline package.json]: Do we need npm publishing? Do we need `private`?

File: src/IMA-CORE.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
+<!-- SPDX-License-Identifier: (AGPL-3.0-only OR CC-BY-4.0) -->
+
+# IMA Agemt Core module
```
Comment (DmytroNazarenko) [inline src/IMA-CORE.md]: Typo

File: src/test/agentUnitTests.js
Comment (DmytroNazarenko) [inline src/test/agentUnitTests.js:1]: Maybe let's move this dir to root test folder?

File: src/OBSERVER.md
Comment (DmytroNazarenko) [inline src/OBSERVER.md:1]: Maybe rename it to SNB.md?

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: CODEOWNERS (hunk 1 above)
Comment (sergiy-skalelabs) [reply CODEOWNERS]: Done

File: README.md (hunk 3 above)
Comment (sergiy-skalelabs) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (sergiy-skalelabs) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (sergiy-skalelabs) [reply README.md]: Done

File: package.json (hunk 6 above)
Comment (sergiy-skalelabs) [reply package.json]: Done

File: src/IMA-CORE.md (hunk 7 above)
Comment (sergiy-skalelabs) [reply src/IMA-CORE.md]: Done

File: src/test/agentUnitTests.js
Comment (sergiy-skalelabs) [reply src/test/agentUnitTests.js:1]: Yes, agree

File: src/OBSERVER.md
Comment (sergiy-skalelabs) [reply src/OBSERVER.md:1]: Done

File: .gitmodules (hunk 2 above)
Comment (sergiy-skalelabs) [reply .gitmodules]: Done

File: src/test/agentUnitTests.js
Comment (sergiy-skalelabs) [reply src/test/agentUnitTests.js:1]: Done

**Review by DmytroNazarenko — APPROVED** [review]

**Review by dmytrotkk — APPROVED** [review]

**Pull request merged**